exercises/palindrome-products/uPalindromeProductsTest.pas

```diff
@@ -0,0 +1,183 @@
+unit uPalindromeProductsTest;
+
+interface
+uses
+  DUnitX.TestFramework, uPalindromeProducts;
+
+const
+  CanonicalVersion = '1.1.0';
+
+type
+  [TestFixture]
+  TPalindromeProductsTest = class(TObject)
+  private
+    Expected, Actual : TPalindromeResult;
+    procedure Compare(Expected, Actual : TPalindromeResult);
+    procedure CompareArrays(Array1, Array2: TArray<integer>);
+
+  public
+    [Test]
+//    [Ignore('Comment the "[Ignore]" statement to run the test')]
+    procedure finds_the_smallest_palindrome_from_single_digit_factors;
+
+    [Test]
+    [Ignore]
+    procedure finds_the_largest_palindrome_from_single_digit_factors;
+
+    [Test]
+    [Ignore]
+    procedure find_the_smallest_palindrome_from_double_digit_factors;
+
+    [Test]
+    [Ignore]
+    procedure find_the_largest_palindrome_from_double_digit_factors;
+
+    [Test]
+    [Ignore]
+    procedure find_smallest_palindrome_from_triple_digit_factors;
+
+    [Test]
+    [Ignore]
+    procedure find_largest_palindrome_from_triple_digit_factors;
+
+    [Test]
+    [Ignore]
+    procedure find_smallest_palindrome_from_four_digit_factors;
+
+    [Test]
+    [Ignore]
+    procedure find_largest_palindrome_from_four_digit_factors;
+
+    [Test]
+    [Ignore]
+    procedure error_result_for_smallest_if_min_is_more_than_max;
+
+    [Test]
+    [Ignore]
+    procedure error_result_for_largest_if_min_is_more_than_max;
+
+    [Test]
+    [Ignore]
+    procedure empty_result_for_smallest_if_no_palindrome_in_the_range;
+
+    [Test]
+    [Ignore]
+    procedure empty_result_for_largest_if_no_palindrome_in_the_range;
+  end;
+
+implementation
+
+uses
+  System.SysUtils;
+
+procedure TPalindromeProductsTest.CompareArrays(Array1, Array2: TArray<integer>);
+var
+  i: integer;
+begin
+  Assert.AreEqual(Length(Array1), Length(Array2), ' - Array lengths must be equal');
+  for i := Low(Array1) to High(Array1) do
+    Assert.AreEqual(Array1[i], Array2[i], format('Expecting element %d to = %d, Actual = %d',
+      [i, Array1[i], Array2[i]]));
+end;
+
+procedure TPalindromeProductsTest.empty_result_for_largest_if_no_palindrome_in_the_range;
+begin
+  Assert.WillRaise(procedure()
```

Comment:
Use `WillRaiseWithMessage` instead.  Just add `WithMessage` after `WillRaise`, everything else is fine.  Repeat this change for all other occurrences.